model/task_history.go

```diff
@@ -336,7 +336,7 @@ func (iter *taskHistoryIterator) GetChunk(v *Version, numBefore, numAfter int, i
 		{"$group": groupStage},
 		{"$sort": bson.M{task.RevisionOrderNumberKey: -1}},
 	}
-	agg := database.C(task.Collection).Pipe(pipeline).MaxTime(taskHistoryMaxTime)
+	agg := db.ConstructPipeWithMaxTime(database, task.Collection, pipeline, taskHistoryMaxTime)
 	var aggregatedTasks []bson.M
 	if err = agg.All(&aggregatedTasks); err != nil {
```

Comment:
I meant rather that here we should call AggregateWithMaxTime (which prevents us from needing ConstructPipeWithMaxTime)

Reply:
I'm maybe confused, but we could just replace 339 through 341 with AggregateWithMaxTime, right? Basically I'm advocating not to have ConstructPipeWithMaxTime since I don't think it's really a helpful helper function.

Reply:
Ok, if we go with that route then before line 346 we will need to reconstruct the same aggregation `agg := database.C(task.Collection).Pipe(pipeline).MaxTime(taskHistoryMaxTime)`. That's why I thought it might be redundant to call `AggregateWithMaxTime` in this function but I can just do that and revert all the helper creations.

Reply:
This solution above broke some unit tests and had me scratching my head for a bit but eventually I got this to work by having `AggregateWithMaxTime ` return both an aggregation as well as an error.  Tests broke because `GetFailedTests ` expects the `Cursor` on the aggregation to be non-nil when it is passed in, which is not the case if we just define a whole new aggregation. It needs to receive the Aggregation in a state after `.All()` has been called on it.